src/main/java/org/openrewrite/java/testing/easymock/EasyMockVerifyToMockitoVerify.java

```diff
@@ -0,0 +1,135 @@
+    private static final MethodMatcher EASY_MATCHER = new MethodMatcher("org.easymock.EasyMock expect(..)");
+
+    @Override
+    public String getDisplayName() {
+        return "Replace EasyMock verify calls with Mockito verify calls";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Replace EasyMock.verify(dependency) with individual Mockito.verify(dependency).method() calls based on expected methods.";
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return Preconditions.check(new UsesMethod<>(VERIFY_MATCHER), new JavaIsoVisitor<ExecutionContext>() {
+            @Override
+            public J.MethodDeclaration visitMethodDeclaration(J.MethodDeclaration method, ExecutionContext ctx) {
+                J.MethodDeclaration md = super.visitMethodDeclaration(method, ctx);
+                if (md.getBody() == null) {
+                    return md;
+                }
+
+                maybeAddImport("org.mockito.Mockito", "verify");
+                maybeRemoveImport("org.easymock.EasyMock.verify");
+
+                int idx = 0;
+                for (Statement statement : md.getBody().getStatements()) {
+                    if (statement instanceof J.MethodInvocation) {
+                        J.MethodInvocation m = (J.MethodInvocation) statement;
+                        if (VERIFY_MATCHER.matches(m) && m.getArguments().size() == 1 && m.getArguments().get(0) instanceof J.Identifier) {
+                            J.Identifier dependency = (J.Identifier) m.getArguments().get(0);
+                            List<Statement> statementsAboveVerify = md.getBody().getStatements().subList(0, idx);
+                            List<J.MethodInvocation> expectedCalls = getExpectedCalls(dependency, statementsAboveVerify);
+
+                            for (int i = 0, expectedCallsSize = expectedCalls.size(); i < expectedCallsSize; i++) {
+                                J.MethodInvocation expectedMethod = expectedCalls.get(i);
+                                List<Expression> parameters = expectedMethod.getArguments();
+                                if (parameters.size() == 1 && parameters.get(0) instanceof J.Empty) {
+                                    parameters.clear();
+                                }
+                                String anyArgs = join(",", nCopies(parameters.size(), "#{any()}"));
+                                parameters.add(0, dependency);
+                                Statement currStatement = md.getBody().getStatements().get(idx);
+                                JavaCoordinates coordinates = i == 0 ? currStatement.getCoordinates().replace() : currStatement.getCoordinates().after();
+                                md = JavaTemplate.builder("verify(#{any()})." + expectedMethod.getSimpleName() + "(" + anyArgs + ")")
+                                        .contextSensitive()
```

Comment:
The template doesn't look like it would have to be context-sensitive, but this is always a bit hard to tell.

Reply:
It seems it does. If I remove it, the `maybeRemoveImport("org.easymock.EasyMock.verify");` does not work anymore 😞.